plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/aggregation/ImplementSum.java

```diff
@@ -54,7 +55,9 @@ public ImplementSum(Function<DecimalType, Optional<JdbcTypeHandle>> decimalTypeH
     @Override
     public Pattern<AggregateFunction> getPattern()
     {
-        return basicAggregation()
+        return Pattern.typeOf(AggregateFunction.class)
+                .with(hasSortOrder().equalTo(false))
+                .with(hasFilter().equalTo(false))
```

Comment:
before this change were we returning incorrect results for sum with filter with and without pushdown? `sum(x) FILTER (WHERE x < 100)`

Reply:
`basicAggregation()` rejects those with filter